### DISCUSSION TITLE
large-language-models-training/

### DISCUSSION THREAD

**Discussion by giscus** [amir-pourmand/hugo-papermod-farsi-template#32] [Announcements]
# large-language-models-training/

فکر می‌کنم حتی اگر اخبار هوش مصنوعی رو دنبال نکرده باشید، راجع به ChatGPT حداقل یک بار شنیدید.
زمانی ترند روز بیت‌کوین و بلاک چین بود و می‌خواستند همه رو چی با بلاک چین انجام بدن، الان ترند ChatGPT و به طور عام‌تر مدل‌های زبانی بزرگه. چیزی که علمای دیپ لرنینگ به طور خلاصه میگن LLM؛ مخفف Large Language Models.
بالاخره از قدیم‌الایام سعی داشتند که بتوانند زبان‌های بشری رو مدل‌سازی کنند و بفهمند.

https://aprd.ir/large-language-models-training/

**Comment by hiwasham** [comment]
امیر یه سری نهادها (یه ویژه نهادهای حکومتی) دنبال این هستن که سیستمهای LLM داخلی رو توسعه بدن. احتمالاً روی LLM های اپسن سورس.
خودشون میگن هم میخوان تجربه اجراش رو داشته باشن هم میترسن از اینکه در شرایط خاصی سرویس شون رو قطع کنن پس میکنم باید سیستم خودمون رو بسازیم.
من به شخصه از دور به پروژه های بومی بسازی (که همشون دولتی ان تا جایی که من دیدم) بدبینم. 
دوست داشتم نظر تو رو در این مورد بدونم
اینکه چقدر شدنیه به نظرت
چقدر توانش رو دارن
آیا لزومی داره یا نه
میدونم اینها جوابهای مشخصی نداره ولی بهانه ای هستن برای گفتن نظر و تجربه ت

**Reply by pourmand1376** [reply]
سلام هیوا جان. 

من قبلاً [یه پستی تو لینکداین](https://www.linkedin.com/posts/amir-pourmand_nemotron-3-8b-chat-steerlm-nvidia-ngc-activity-7147363427387166720-KRfc) راجع به این موضوع گذاشتم و اشاره‌ای کردم ولی خب اینجا راحت‌ترم که صحبت کنم. 

من هم با شما هم نظرم که این پروژه‌ها بیشتر برای بودجه گرفتن و ژست پروژه بومی گرفتن خوبه و تو عمل کاربرد چندانی نخواهد داشت. 

من فکر می‌کنم ما منابع کافی برای آموزش همچین مدل‌هایی رو نداریم. منابع که میگم منظورم هم متخصص‌اش هست که کمتر کسی تو دنیا می‌دونه واقعاً داره چیکار می‌کنه و تجربه آموزش‌دادن مدل‌های انقدر بزرگ رو داره و هم منابع سخت‌افزاری و نرم‌افزاری. الان شرکت‌هایی که تو خارج کشور دارند مدل زبانی پایه آموزش میدن صدها و بلکه هزاران GPU قدرتمند دارند. کدوم ارگانی تو کشور همچین منابعی رو می‌تونه تأمین کنه؟

تا اونجا که من می‌دونم از یکی از نهاد‌های بزرگ لطف کردند ۸ تا GPU به یک شرکت دانش‌بنیان دادند و گفتند ۶ ماه دیگه می‌خوایم یکی مثل ChatGPT داشته باشیم!

با شنیدن این حرف‌ها یاد موتور جستجوگر بومی میفتم که نتیجه‌اش رو خودتون دیدید. اتفاقاً همین چند روز پیش یک تیم بزرگ از محققان از سرتاسر دنیا مدلی رو با پشتیبانی از ۱۰۱ زبان (از جمله فارسی) ارائه کردند و خود مدل و دیتاست رو هم رایگان و متن‌باز منتشر کردند. 

الان شرایط کاملاً برای یک شرکت «دانش بنیان» متشکل از «نخبگان» محیاست که همین رو کمی تغییر بدن و یه دستی به سر و روش بکشن و بهشون تحویل بدند. 

البته این روزها اکثر شرکت‌ها (چه ایرانی و چه خارجی) دوست دارند که مدل زبانی خاص خودشون رو داشته باشند. هم از نظر هزینه ممکنه براشون به صرفه‌تر باشه (چون ممکنه GPU بیکار داشته باشند) و هم از نظر محرمانگی اطلاعات و حفظ امنیت داده و چیزهایی از این دست. 

بعید می‌دونم حتی تو خود آمریکا هم به شرکت OpenAI کاملاً اعتماد کنند و محرمانه‌ترین اطلاعات رو به انواع و اقسام روش‌های مختلف در اختیار شرکت قرار بدن به این امید که ان‌شاءالله OpenAI به حرف‌هاش عمل کنه و روی این داده‌ها Training انجام نده و ذخیره‌شون هم نکنه. 

ولی باز هم این حرفا دلیل نمیشه که مدل ان‌شاءالله ماشاءالله و هیئتی بخوایم روی چیزی که منابع کافی براش نداریم، سرمایه‌گذاری کنیم. به نظرم بهترین روش برای این‌طور کارها مشارکت متن‌باز هست که خیرش به همه می‌رسه و تهش هم واقعاً ازش یه کار مناسب درمیاد. مشابه همین کاری که تیم Cohere برای مدل Aya-101 انجام دادند. 

پی‌نوشت: پیامت چند روزه جلوی چشممه و می‌خوام جواب بدم ولی هر بار غولی به نام کمال‌گرایی میاد و میگه بذار سرفرصت کامل‌تر جواب بده.

**Reply by hiwasham** [reply]
موافقم با صحبتهات. تعدادی از این پروژه هارو از نزدیک میدیدم. از جمله همین موتور جستجوی ملی. چیزی شبیه جوک بود. روزهای لانچ شون هم یادمه اتفاقات عجیبی افتاد.
سال 90 خودم هم عضو تیم روتر UT بودم. تیمی که قرار بود اولین روتر کشور رو بسازه با 7-8 میلیارد بودجه (که اون موقع عدد بدی نبود) ولی نتیجه ش با اینکه از نظر دانشگاه، قابل قبول بود (یه روتر یک ترابیت بر ثانیه) اما در مقایسه با استاندارد جهانی، شبیه جوک بود. با چند صد هزار تومن میشد یه محصول بهتر رو از مارکت رو خرید. دانش و تجربه ش هم زیاد نمیارزه مخصوصا اگه بدونیم مرتبط هست به برادران سبزپوشی که میامدن و میرفتن. 
- یکی از دوستام که درگیر پروژه مربوط به LLM هست از یه بودجه 200 میلیاردی صحبت میکرد واسه پردازنده هایی که نیاز دارن ولی در مجموع به نظرم وقتی از نظر تیمی و سیستمی، این فضا سالم نیست، اون انرژی سازنده ای که آدمها و تیم وارد کار میکنه وجود نداره برای همین چیز تمیزی ازش در نمیاد. همه چی مصنوعی و زوری و سفارشیه. 
- مسئله محرمانگی و تحریم های احتمالی آینده به نظرم تنها نکته ای هست که میتونن روش مانور بدن و کار روی چنین پروژه هایی رو توجیه کنن. البته میدونیم که این پروژه ها هم روی کار اپن سورس بقیه انجام میشه. وگرنه بدیهیه که توان وارد شدن به این فضا رو ندارن. اما ادعاشون در حد کسیه که پدر مادر این حوزه ست :))
- من چند دقیقه ای تایپ کردم شرایطم اجازه کمال گرایی رو بهم نمیده. کلاً در حال دویدن هستم :))
